src/main/webapp/main.js

```diff
@@ -58,14 +62,43 @@ Promise.all([mapPromise, domPromise, firebasePromise]).then(() => {
         location.href = "/";
       }
 
+      userId = firebase.auth().currentUser.uid;
       getServerUrl().then(result => {
         connection = new WebSocket(result);
         initWebsocket();
-        myMap = new ChapMap();
+        
+        getCoords().then(coords => {
+          myMap = new ChapMap(coords);
+        }).catch(() => {
+          myMap = new ChapMap([DEFAULT_LAT, DEFAULT_LNG]);
+        })
+       
         initChat();
       });
     });
 
+/**
+ * Returns the user's coordinates, if possible
+ * @returns{!Object} Tuple representing the user's latitude and longitude
+ */
+function getCoords(){
+
+  return new Promise(function(resolve, reject){
+    
+    if(navigator.geolocation){
+      navigator.geolocation.getCurrentPosition(function(position){
+        if(position.coords.latitude !== undefined){
```

Comment:
nit: `if(position.coords.latitude)` is equivalent to `if(position.coords.latitude !== undefined)`

Reply:
Gotcha! I'll make that fix. Thanks for catching that!